activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -103,6 +103,10 @@ def supports_virtual_columns?
         mariadb? || version >= "5.7.5"
       end
 
+      def supports_optimizer_hints?
+        !mariadb? && version >= "5.7.7"
+      end
```

Comment:
Users of MariaDB and older MySQL will confuse these for index hints support. Nothing we can do here, except clarify it in documentation, I think.

Reply:
Or add support for index hints, too :grin:

Reply:
MySQL/MariaDB, SQL Server, and SQLite3 has a (table qualified) index hints feature.
Unlike optimizer hints, any table qualified hints is easily used by `from` method for a long time (e.g. `relation.from("topics FORCE INDEX(PRIMARY)")`).

I'd like to keep the scope of this PR as much as small...

Reply:
Added just a comment `See https://dev.mysql.com/doc/refman/8.0/en/optimizer-hints.html for more details.`
https://github.com/rails/rails/pull/35615/files#diff-98ce5dc0cd01f8b0435ff6feac10d8a8R106